src/HostRouter.ts

```diff
@@ -24,14 +24,15 @@ export class HostRouter {
    *
    * @param route The route to lookup, such as '/foo/bar/baz'
    */
-  public getClientTarget(route: string): ClientTarget {
-    let clientTarget: ClientTarget = {
-      id: null,
-      url: null,
-      assignedRoute: null
-    };
+  public getClientTarget(route: string): ClientTarget | null {
+    const desiredRoute = route || '';
```

Comment:
With the compiler guarantee that `route` is always a string, you no longer need the intermediate `desiredRoute` variable for handling null values.